File: Backend/src/main/java/com/example/PublicKeyInfrastructure/dto/certificate/IntermediateCertificateDTO.java
package com.example.PublicKeyInfrastructure.dto.certificate;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class IntermediateCertificateDTO {

    private int issuerCertificateId;
    private String subjectCommonName;
    private String subjectOrganizationName;
    private String subjectOrganizationalUnit;
    private String subjectCountry;
    private String subjectEmail;
    private String serialNumber;
    private LocalDateTime validFrom;
    private LocalDateTime validTo;
    private boolean isDigitalSignature;
    private boolean isKeyEncipherment;
    private boolean isServerAuth;
    private boolean isClientAuth;
    private boolean isAdminCreating;
}

File: Frontend/src/dto/certificate/IntermediateCertificateDTO.ts
export interface IntermediateCertificateDTO {
    issuerCertificateId: number,
    subjectCommonName: string,
    subjectOrganizationName: string,
    subjectOrganizationalUnit: string,
    subjectCountry: string,
    subjectEmail: string,
    validFrom: string,
    validTo: string,
    isDigitalSignature: boolean,
    isKeyEncipherment: boolean,
    isServerAuth: boolean,
    isClientAuth: boolean,
    isAdminCreating: boolean
}